java-checks/src/main/java/org/sonar/java/checks/naming/MethodNamedEqualsCheck.java

```diff
@@ -43,25 +41,36 @@
   @Override
   public void visitNode(Tree tree) {
     MethodTree methodTree = (MethodTree) tree;
-    if ("equals".equalsIgnoreCase(methodTree.simpleName().name()) && !hasSingleObjectParameter(methodTree) && !isOverriding(methodTree)) {
-      reportIssue(methodTree.simpleName(), "Either override Object.equals(Object), or totally rename the method to prevent any confusion.");
+    if (equalsWithSingleParam(methodTree) && !hasProperEquals(methodTree)) {
+      reportIssue(methodTree.simpleName(), "Either override Object.equals(Object), or rename the method to prevent any confusion.");
     }
   }
 
-  private static boolean hasSingleObjectParameter(MethodTree methodTree) {
-    List<VariableTree> parameters = methodTree.parameters();
-    if (parameters.size() != 1) {
-      return false;
-    }
-    return isObjectType(parameters.get(0));
+  private static boolean equalsWithSingleParam(MethodTree methodTree) {
+    return "equals".equalsIgnoreCase(methodTree.simpleName().name()) && methodTree.parameters().size() == 1;
   }
 
-  private static boolean isObjectType(VariableTree variableTree) {
-    String type = ExpressionsHelper.concatenate((ExpressionTree) variableTree.type());
-    return "Object".equals(type)|| "java.lang.Object".equals(type);
+  private static boolean hasProperEquals(MethodTree methodTree) {
+    Symbol.MethodSymbol symbol = methodTree.symbol();
```

Comment:
This can only happen if semantic is not present (symbol method is not annotated as returning null). 
As such you should test if semantic is present and not raise an issue if it is not the case.